### PR TITLE
New Major Version 3.0.0 - OSCReceiver refactor and new features

### PR DESCRIPTION
**Pull request by SteffeyDev**
This was actually really fun, I refactored the way the receiver works to use pattern-matching and callbacks that are built into a tree structure for fast runtime matching.  This provides a much easier way to add methods in the future, has some auto-documenting benefits, allows for better type-casting support, and brings implicit TouchOSC backward-compatibility to all methods.

Because this is such a major change to the underlying code, it also involved removing many deprecated methods, and thus warrants a new major version.

Included in these changes are DSK parameter methods (#130) and fully documented HyperDeck control (#58).  It also has some potential crash fixes.

I'm looking for code review and testing before merging.  I've done a fair amount of testing but included the binary as always for others to test.

### CODE REVIEW

**Comment by randallpacker** [comment]
Thanks Peter, this sounds incredible! As I am a user rather than a coder, is it possible you can post an installer of the latest version so I can test? Best, Randall

**Comment by SteffeyDev** [comment]
@randallpacker you can download the ZIP and run the atemOSC.app in the unzipped folder: https://github.com/SteffeyDev/atemOSC/archive/reciever-refactor.zip

Thanks for jumping in and offering to test!

**Comment by SteffeyDev** [comment]
Anyone remember how to get signing working for travis?

File: atemOSC/atemOSC.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -209,7 +551,8 @@
 				LastUpgradeCheck = 0830;
 				TargetAttributes = {
 					8D1107260486CEB800E47090 = {
-						DevelopmentTeam = K5C6E7A2D6;
+						DevelopmentTeam = LRNWZB2D4D;
```
Comment (danielbuechele) [inline atemOSC/atemOSC.xcodeproj/project.pbxproj]: This is what breaks code signing

**Comment by danielbuechele** [comment]
Wow, that's a really cool refactoring. Thanks for working on this. I agree this should be a new major release. The code looks way cleaner than before. Feel free to merge, when you think it's ready.

File: atemOSC/OSCAddressPanel.mm [hunk 2]
```diff
@@ -198,45 +208,76 @@ - (void)setupWithDelegate:(AppDelegate *)appDel
...
+						address = [address stringByAppendingString:@" <string>"];
+					[self addEntry:label forAddress:address toString:helpString];
+				}
+			}
 		}
 	}
+	
+	[helpString appendAttributedString:[[NSAttributedString alloc] initWithString:@"\n\nNote: Additional addresses are available that provide backward-compatibility with TouchOSC.  See the Readme on Github for details.\n\n"]];
+	
+	[helpString appendAttributedString:[[NSAttributedString alloc] initWithString:@"We add support for address on an as-needed basis.  If you are in need of an additional addresses, open an issue on Github letting us know what it is.\n"]];
```
Comment (sneat) [inline atemOSC/OSCAddressPanel.mm]:
```suggestion
	[helpString appendAttributedString:[[NSAttributedString alloc] initWithString:@"We add support for addresses on an as-needed basis.  If you are in need of an additional address, open an issue on Github letting us know what it is.\n"]];
```

**Comment by randallpacker** [comment]
@SteffeyDev @danielbuechele I just tested the beta version 3.0 with every single command I am using with both the ATEM and Hyperdecks (which is most everything) and it works like a charm (so far). The only funny thing that happens is when you launch the app it ask if you want to download the new version, so I just skipped that since I was already opening the new version. Bravo!!!

File: README.md [hunk 3]
```diff
@@ -204,67 +225,4 @@ Feedback: None
    - Get the Maximum Number of Macros: `/atem/macros/max-number`
      - Returns an `int` of the maximum number of Macros supported by your ATEM
      - Access to these Macros should be used via an index of `n-1`
-   - Stop the currently active Macro (if any): `/atem/macros/stop`
```
[the remark this answers is not in the document]
Comment (SteffeyDev) [reply README.md]: Good catch, looks like the github web editor bugged out on my last update.

**Comment by SteffeyDev** [comment]
Ok, I standardized the README spacing and format and fixed the cut-off

**Comment by danielbuechele** [comment]
Ok, one super minor thing. For the updated screenshot, can we make one without the wallpaper visible? (Press the space bar, when in screenshot mode and click on the window)

File: atemOSC/atemOSC.xcodeproj/project.pbxproj (hunk 1 above)
Comment (SteffeyDev) [reply atemOSC/atemOSC.xcodeproj/project.pbxproj]: Thanks, I did some rebasing and remove unnecessary changes to that file.

**Comment by SteffeyDev** [comment]
> Ok, one super minor thing. For the updated screenshot, can we make one without the wallpaper visible? (Press the space bar, when in screenshot mode and click on the window)

Due to the needed signing updates, I can't actually run the app right now, and also don't have a switcher available to get a better log message.  @danielbuechele Can I leave the screenshot for now and then we can update it later.  It is the default 10.15 Catalina background, so should be familiar to users.

**Comment by SteffeyDev** [comment]
@randallpacker Thanks so much for testing! That makes me feel way better about releasing.

**Comment by danielbuechele** [comment]
100%! Leave the screenshot as it is for now.

**Comment by SteffeyDev** [comment]
Ok, thanks! merging and then I'll tag the commit for deployment

**Comment by danielbuechele** [comment]
Awesome! Congrats and again, thanks for working on this.

Should we remove the actual app from repo as it can be downloaded from the release page?

**Pull request merged**